src/Servers/Kestrel/Core/src/Features/IWebTransportSession.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features;
+
+/// <summary>
+/// Controls the session and streams of a WebTransport session.
+/// </summary>
+public interface IWebTransportSession
+{
+    /// <summary>
+    /// The id of the WebTransport session.
+    /// </summary>
+    public long SessionId { get; }
+
+    /// <summary>
+    /// Abruptly close the WebTransport session and stop all the streams.
+    /// </summary>
+    /// <exception cref="Exception">If This is not a valid WebTransport session.</exception>
+    public void Abort();
+
+    /// <summary>
+    /// Returns the next incoming stream in the order which Kestel received it. The stream can be either bidirectional or unidirectional.
+    /// </summary>
+    /// <exception cref="Exception">If this is not a valid WebTransport session or it fails to get a stream to accept.</exception>
+    /// <param name="cancellationToken">The cancellation token used to cancel the operation.</param>
+    /// <returns>The unidirectional or bidirectional stream that is next in the queue.</returns>
+    public ValueTask<Stream> AcceptStreamAsync(CancellationToken cancellationToken);
```

Comment:
What happens to a pending accept call if the session ends? How about having it return null.
```suggestion
    public ValueTask<Stream?> AcceptStreamAsync(CancellationToken cancellationToken);
```

Reply:
Would it be better to throw in this case? Since streams can only exist during a session, if the session ends, shouldn't we notify the application that something bad is happening?